core/new-gui/src/app/workspace/component/workflow-editor/workflow-editor.component.ts

```diff
@@ -1,15 +1,151 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, AfterViewInit } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import '../../../common/rxjs-operators';
 
+import * as joint from 'jointjs';
+import { JointUIService } from '../../service/joint-ui/joint-ui.service';
+
+/**
+ * WorkflowEditorComponent is the componenet for the main workflow editor part of the UI.
+ *
+ * This componenet is binded with the JointJS paper. JointJS handles the operations of the main workflow.
+ * The JointJS UI events are wrapped into observables and exposed to other components / services.
+ *
+ * See JointJS documentation for the list of events that can be captured on the JointJS paper view.
+ * https://resources.jointjs.com/docs/jointjs/v2.0/joint.html#dia.Paper.events
+ *
+ * @author Zuozhi Wang
+ * @author Henry Chen
+ *
+*/
 @Component({
   selector: 'texera-workflow-editor',
   templateUrl: './workflow-editor.component.html',
   styleUrls: ['./workflow-editor.component.scss']
 })
-export class WorkflowEditorComponent implements OnInit {
+export class WorkflowEditorComponent implements AfterViewInit {
+
+  // the DOM element ID of the main editor. It can be used by jQuery and jointJS to find the DOM element
+  // in the HTML template, the div element ID is set using this variable
+  public readonly WORKFLOW_EDITOR_JOINTJS_WRAPPER_ID = 'texera-workflow-editor-jointjs-wrapper-id';
+  public readonly WORKFLOW_EDITOR_JOINTJS_ID = 'texera-workflow-editor-jointjs-body-id';
+
+  public paper: joint.dia.Paper = null;
+  // this is only the temporary place for the graph variable
+  // it will be moved to a service in subsequent PRs
+  public graph: joint.dia.Graph = new joint.dia.Graph();
+
+  constructor(
+    private jointUIService: JointUIService
+  ) { }
+
+  ngAfterViewInit() {
+
+    this.createJointjsPaper();
+
+    // add a 500ms delay for joint-ui.service to fetch the operator metaData
+    // this code is temporary and will be deleted in future PRs when drag
+    // and drop is implemented
+
+    Observable.of([]).delay(500).subscribe(
+      emptyData => {
+        // add some dummy operators and links to show that JointJS works
+        this.graph.addCell(
+          this.jointUIService.getJointjsOperatorElement(
+            'ScanSource',
```

Comment:
Why are these operators being added? Is it just for this PR?

Reply:
This is just for this PR and will be deleted in the future. This operators show that the custom element defined is successfully registered and being utilized by JointJS